pkg/yurtmanager/webhook/pod/v1alpha1/pod_default.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the License);
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/openyurtio/openyurt/pkg/apis/apps"
+)
+
+// Default implements builder.CustomDefaulter.
+func (webhook *PodHandler) Default(ctx context.Context, obj runtime.Object, req admission.Request) error {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a Pod but got a %T", obj))
+	}
+
+	// Add NodeAffinity to pods in order to avoid pods to be scheduled on the nodes in the hostNetwork mode NodePool
+	annotations := pod.ObjectMeta.GetAnnotations()
+	if annotations != nil && annotations[apps.AnnotationExcludeHostNetworkPool] == "true" {
+		pod.Spec.Affinity = &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+					NodeSelectorTerms: []corev1.NodeSelectorTerm{
+						{
+							MatchExpressions: []corev1.NodeSelectorRequirement{
+								{
+									Key:      "nodepool.openyurt.io/hostnetwork",
+									Operator: corev1.NodeSelectorOpNotIn,
+									Values:   []string{"true"},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+	}
```

Comment:
we should not overwrite the existing `pod.Spec.Affinity`, this means we need to append affinity only.

Reply:
that's right, fixed.